src/test/fuzz/i2p.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <common/args.h>
+#include <i2p.h>
+#include <netaddress.h>
+#include <netbase.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/fuzz/util/net.h>
+#include <test/util/setup_common.h>
+#include <util/fs_helpers.h>
+#include <util/threadinterrupt.h>
+
+void initialize_i2p()
+{
+    static const auto testing_setup = MakeNoLogFileContext<>();
+}
+
+FUZZ_TARGET(i2p, .init = initialize_i2p)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+
+    SetMockTime(ConsumeTime(fuzzed_data_provider));
```

Comment:
Does this mean that the time will be frozen for the duration of the test?

From the OP:

> GetTime is called at points in i2p

Where is `GetTime()` called in i2p?

Reply:
> Where is GetTime() called in i2p?

It's used in `Sock` (e.g. `RecvUntilTerminator`).

Reply:
Yes, `GetTime()` is also called in `SendComplete`.

> Does this mean that the time will be frozen for the duration of the test?

I believe so, unless I'm missing something. For each iteration, `g_mock_time` will be set to an integer within the range specified in `ConsumeTime`.

Reply:
The frozen time will defeat `FUZZED_SOCKET_FAKE_LATENCY`:

https://github.com/bitcoin/bitcoin/blob/91e0beede2859dea987ba6db746e95dca0ceb024/src/test/fuzz/util/net.cpp#L229-L231

`RecvUntilTerminator()` and `SendComplete()` will never timeout. Is this acceptable trade off in order to have the fuzz test "more stable"? If it is "unstable" without freezing the time, does that mean that some of those methods actually timed out, resulting in a wider coverage? In this aspect, it seems that stable = only covering the non-timeout-codepath = narrower coverage = worse.

Would it make sense to use mock time, but advance it a little in each `Recv()` or `Send()` method?

Reply:
I think it's an acceptable tradeoff, as using mock time makes the test more determinstic. I don't think timeouts are too critical a part of the I2P code, and this harness as is reaches almost all of the code paths in `i2p`. 

But I see what you're saying in that advancing time would be a more accurate simulation. Maybe we leave this as a potential follow up on `FuzzedSock`?

Reply:
> I think it's an acceptable tradeoff ...

Ok.

> ... advancing time would be a more accurate simulation. Maybe we leave this as a potential follow up ...

Right, further changes in `FuzzedSock` itself are kind of out of scope for this PR which resurrects the previously deleted test.

Thanks!